lib/data_pull.rb

```diff
@@ -198,6 +206,24 @@ def run(args:, config:)
         json: output,
       )
     end
+
+    private
+
+    def compute_requesting_issuers(uuids)
+      service_providers = ServiceProviderIdentity.where(uuid: uuids).pluck(:service_provider)
+      return nil if service_providers.empty?
+      service_provider = service_providers.tally.max_by { |_sp, count| count }[0]
+
+      if service_providers.count > 1
+        warn "Multiple computed service providers: #{service_providers.join(', ')}"
+      end
+
+      warn "Computed service provider #{service_provider}"
+
+      [
+        service_provider,
+      ]
```

Comment:
I'd just oneline this?
```suggestion
      Array(service_provider)
```